File: src/main/java/com/freshsip/freshsip/service/ReviewService.java
package com.freshsip.freshsip.service;



import com.freshsip.freshsip.dto.ReviewDTO;
import com.freshsip.freshsip.dto.UserDTO;
import com.freshsip.freshsip.entity.Review;
import org.springframework.data.jpa.repository.Query;

import java.util.List;

public interface ReviewService {
    void addReview(ReviewDTO reviewDTO);

    List<ReviewDTO> getAllReviews();

    ReviewDTO getReviewById(Long reviewId);

    boolean deleteReviewById(Long reviewId);

    String updateReviewById(Long reviewId, Review reviewDTO);
}

File: src/Service/ReviewService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/FreshSip';
class ReviewService {
    submitReview(reviewData) {
        return axios.post(`${API_URL}/user/review`,reviewData); 
    }
    getReviewById(reviewID) {
        return axios.get(`${API_URL}/user/${reviewID}`);
    }

    updateReview(reviewID,reviewData) {
        return axios.put(`${API_URL}/user/${reviewID}`,reviewData);
    }

    deleteReviewById(reviewID) {
        return axios.delete(`${API_URL}/user/${reviewID}`);
    }
}

export default new ReviewService();